pkg/kubectl/cmd/logs.go

```diff
@@ -162,18 +175,34 @@ func (o *LogsOptions) Complete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Com
 	o.ClientMapper = resource.ClientMapperFunc(f.ClientForMapping)
 	o.Out = out
 
+	if selector != "" {
+		if logOptions.Follow {
+			return cmdutil.UsageError(cmd, "only one of follow (-f) or selector (-l) is allowed")
+		}
+		if logOptions.Container != "" {
+			return cmdutil.UsageError(cmd, "a container cannot be specified when using a selector (-l)")
+		}
+		if logOptions.TailLines == nil {
+			logOptions.TailLines = &selectorTail
```

Comment:
We can allow this. I was just follow what was described in the issue. I am assuming the concern was that your selector may match pods that are not guaranteed to have the matching container. One could choose to just ignore pods that do not have the container, although the current code would display an error. So you would have different behavior either one the parsing or on the output.